unfeaturized slots (#6817)
RasaHQ/rasa: docs/docs/domain.mdx

as well as information gathered about the outside world (e.g. the result of a
database query).

Slots are defined in the `slots` section of your domain with their name and
[type](domain.mdx#slot-types). The slot type will determine the data type of the slot
and how the slot
[influences the assistant's behavior](domain.mdx#slots-and-conversation-behavior).

> The following example defines a slot with name `slot_name` and the type `text`.

Reviewer: ```suggestion
The following example defines a slot with name `slot_name` and type `text`.
```

```yaml
slots:

>   # the name of the slot

Reviewer: I think it's clear enough from the sentence before the example
```suggestion
```

  slot_name:

>     # the type of the slot

Reviewer: ```suggestion
```

    type: text
```

### Slots and Conversation Behavior

Most of the time, you want slots to influence how the dialogue progresses.
For example, if your user has provided their home city, you might
have a slot called `home_city`. If the user asks for the
weather, and you *don't* know their home city, you will have to ask

> them for it. A [text slot](domain.mdx#text-slot) only tells Rasa Core whether the slot

Reviewer: not sure about this one:
```suggestion
them for it. A [text slot](domain.mdx#text-slot) only tells Rasa whether the slot
```

Author: changed the whole sentence 🙈

has a value. The specific value of a `text` slot (e.g. Bangalore
or New York or Hong Kong) doesn't make any difference.

```yaml
slots:
  # this slot will influence the conversation depending on
  # whether the slot is set or not
  home_city:
    type: text
    influence_conversation: true
```

> By default all slot types will influence the conversation except slots of the type

Reviewer: ```suggestion
By default all slot types will influence the conversation except slots of type
```

`any`. Depending on the [slot type](./domain.mdx#slot-types), a set slot will cause
different behavior of your assistant so be careful which type you choose.

If you want to store information but don't want a slot to influence the conversation,
set the `influence_conversation` property of your slot to `false`.
The following example defines a slot `age` which will store information about the
user's age, but which will *not* influence the flow of the conversation. This means
that the assistant will ignore the value of the slot during the conversation.

```yaml
slots:
  age:
    type: text
    # this slot will not influence the predictions
    # of the dialogue policies
    influence_conversation: false

> 

Reviewer: ```suggestion
```

```

### Slot Types

> Make this a nice table instead.

Author: I think a table would be too wide and hence annoying. Also, this way we can link to individual slot types which is nice.

        type: text
  ```

* **Description**

  If `influence_conversation` is set to `true`, the assistant's behavior will change

>   depending on whether the slot is set or not. Different texts do not influence the

Reviewer: ```suggestion
  depending on whether the slot is set or not. Different slot values do not influence the
```

Author: I chose `texts` as different slot values _do_ influence the prediction (if the value is `None`).

Reviewer: 👍

        type: bool
  ```

* **Description**

  If `influence_conversation` is set to `true`, the assistant's behavior will change
  depending on whether the slot is set to `true` or `false`. An empty `bool` slot

>   is the same as if the slot was set to `False`.

Reviewer: > is the same as if the slot was set to `False`.

I find this surprising. does this have to be the case? this means a bool slot's default value is always `false`

Author: you can change the `initial_value`  🤔 I think changing this would also be out of the scope of this PR

>   depending on whether the slot is set to `true` or `false`. An empty `bool` slot
>   is the same as if the slot was set to `False`.

Reviewer: ```suggestion
  depending on whether the slot is set to `true` or `false`. An empty `bool` slot
  is the same as if the slot was set to `false`.
```

        - high
  ```

* **Description**

  If `influence_conversation` is set to `true`, the assistant's behavior will change
  depending on the concrete value of the slot. This means the assistant's behavior is
  different depending on whether the slot in the above example has the value `low`,
  `medium`, or `high`.

  A default value `__other__` is automatically added to the user-defined
  values. All values encountered which are not explicitly defined in the slot's `values`
  are mapped to `__other__`.
  `__other__` should not be used as a user-defined value; if it is, it
  will still behave as the default to which all unseen values are mapped.

#### Float Slot

* **Type**

  `float`

* **Use For**

>   Storing continuous numbers.

Reviewer: ```suggestion
  Storing real numbers.
```
or just
```suggestion
  Storing numbers.
```

      max_value:  100.0
  ```

* **Defaults**

  `max_value=1.0`, `min_value=0.0`

* **Description**

  If `influence_conversation` is set to `true`, the assistant's behavior will change
  depending on the value of the slot. All values below `min_value` will be treated as
  `min_value`, the same happens for values above `max_value`.
  Hence, if `max_value` is set to `1`, there is no difference between the slot values

>   `2` and `3.5`.

Reviewer 2: How would this slot type be used in a story? Based on the description it sounds like it only matters whether the value is above max or below min, but is that true?

Author: The story will be different depending on the concrete value of the float. So the slot value is capped to the supplied min / max values and then everything within this range is different for the bot. Would writing
```
depending on the value of the number
```
make it clearer?

Reviewer 2: ```suggestion
  If `influence_conversation` is set to `true`, the assistant's behavior will change
  depending on the value of the slot. If the value is between `min_value` and `max_value`, the specific value of the number is used. All values below `min_value` will be treated as
  `min_value`, and all values above `max_value` will be treated as `max_value`.
  Hence, if `max_value` is set to `1`, there is no difference between the slot values
  `2` and `3.5`.
```

better explicit than confused?

      type: list
  ```

* **Description**

  If `influence_conversation` is set to `true`, the assistant's behavior will change
  depending on whether the list is empty or not. The **length of the list stored in
  the slot does not influence the dialogue**.

> 

Reviewer 2: adding this note because it's  not mentioned anywhere
```suggestion

If a `list` slot is [auto filled by an entity of the same name](domain.mdx#slot-auto-fill), each entity value will be appended to the value of the slot. 

```

Reviewer 2: Or it could go under "auto fill slots"

Author: @melindaloubser1 This means if there are previous values stored, they won't be overridden?

Reviewer 2: I believe so? That's what I thought, but one way or another we should document the behaviour

Reviewer 2: Ok nope, that's what [this forum post]() implied, but I checked and it overwrites.
So this type is only really useful for custom actions right?

Author: your link doesn't link to the forum post 😆

Author: Judging from the code it should be replaced and not extended

Reviewer 2: Oh ugh now I can't find it, but yes it seems like that's always been the case

#### Any Slot

* **Type**

  `any`

* **Use For**

>   Storing arbitrary types of values.

Reviewer: ```suggestion
  Storing arbitrary values (they can be objects of any type, such as dictionaries).
```
maybe it's just me, but "type" has a special meaning in slots, and I worry that that is confusing. `type` in this case refers to any legal yaml object, not the slot types previously defined

Author: Is a list an object? 🤔

Reviewer: or just
> they can be of any type, such as dictionaries or lists

* **Example**

  ```yaml-rasa
  slots:
    shopping_items:
      type: any
  ```

* **Description**

>   Slots of type `any` are always ignored during conversations. If you want to store

Reviewer 2: ```suggestion
  Slots of type `any` are always ignored during conversations. The property `influence_conversation` cannot be set to `true` for this slot type. If you want to store
```
